Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -3,7 +3,50 @@ import Foundation
 
 // MARK: - ResourceOptions
 
-extension ResourceOptions {
+/// Options to configure access to a resource
+public struct ResourceOptions {
+
+    /// The access token to access the resource. This should be a valid, non-empty,
+    /// Mapbox access token
+    public var accessToken: String
```

Comment:
I'd like us to consider making access token a custom type, for example an enum. Not unlike an optional, but instead of a `.none` case, we could have a `.search` or `.default` case that would look for the token in the Info.plist etc. This way it's more explicit and we could deprecate it later if we ever choose to.

Reply:
I like this idea.. may be we could even make it `expressibleByString` in some way?

Reply:
cc @Chaoba 
on Android we can not overwrite the type in the ResourceOptions..